Update protocol examples to support all network types #13249 (IDFGH-12196) (#13250)
espressif/esp-idf: examples/protocols/http_request/README.md

```
I (17034) example: DNS lookup v4.ipv6-test.com (port 80)
I (17034) example_common: IPv6 DNS lookup
I (17074) example_common: - IPv6 DNS lookup failed, trying IPv4 lookup
I (17074) example_common: IPv4 DNS lookup
I (17074) example: DNS lookup succeeded. (2) IP=51.75.78.103
I (17084) example: ... allocated socket
I (17694) example: ... connected
I (17694) example: ... socket send success
I (17694) example: ... set socket receiving timeout success
HTTP/1.1 200 OK
Date: Sat, 24 Feb 2024 06:10:00 GMT
Server: Apache/2.4.25 (Debian)
Vary: Accept-Encoding
Connection: close
Content-Type: text/html; charset=UTF-8

220.240.255.134
I (18314) example: ... done reading from socket. Last read return=0 errno=128.
I (18314) example: 10...
```

### IPv4-only network connecting to dual-stack server

This shows an IPv4 only network (Shadow) connecting to a dual-stack server (`v4v6.ipv6-test.com`).

The connection attempt starts as soon as the IPv4 address is received, and as the device has no global IPv6 address, the DNS lookup and connection will only be IPv4 (`51.75.78.103`).

> Note that although the device has an IPv6 link-local address it is ignored, as it is not usable to connect to a public address returned from DNS.

Reviewer: ```suggestion
Note that, although the device has an IPv6 link-local address, it is ignored because it cannot be used to connect to a public address returned from DNS
```

Author: Done. Fix included and rebased to lastest head.
